### DISCUSSION TITLE
Ollama offline connection failed

### DISCUSSION THREAD

**Discussion by start-life** [open-webui/open-webui#1901] [General]
Ollama offline connection failed
When it's online it connects

INFO:     Started server process [8800]
INFO:     Waiting for application startup.
INFO:apps.litellm.main:start_litellm_background
INFO:apps.litellm.main:run_background_process
INFO:apps.litellm.main:Executing command: ['litellm', '--port', '14365', '--host', '127.0.0.1', '--telemetry', 'False', '--config', 'C:\\zxcv\\open-webui\\backend\\data/litellm/config.yaml']
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
INFO:apps.litellm.main:Subprocess started successfully.
INFO:     127.0.0.1:51747 - "GET /api/config HTTP/1.1" 200 OK
INFO:     127.0.0.1:51748 - "GET /manifest.json HTTP/1.1" 200 OK
INFO:     127.0.0.1:51748 - "GET /api/v1/auths/ HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51748 - "GET /ollama/api/tags HTTP/1.1" 200 OK
INFO:apps.openai.main:get_all_models()
INFO:apps.openai.main:get_all_models()
INFO:     127.0.0.1:51748 - "GET /openai/api/models HTTP/1.1" 200 OK
INFO:     127.0.0.1:51748 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/modelfiles/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/prompts/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/documents/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/chats/tags/all HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51754 - "GET /ollama/api/tags HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:51754 - "GET /api/changelog HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:apps.openai.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:apps.openai.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51754 - "GET /openai/api/models HTTP/1.1" 200 OK
INFO:     127.0.0.1:51759 - "GET /ollama/api/version HTTP/1.1" 500 Internal Server Error
INFO:     127.0.0.1:51754 - "GET /static/favicon.png HTTP/1.1" 200 OK
INFO:     127.0.0.1:51759 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51761 - "GET /ollama/urls HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
ERROR:apps.ollama.main:Connection error: Cannot connect to host localhost:11434 ssl:default [getaddrinfo failed]
INFO:     127.0.0.1:51761 - "GET /ollama/api/version HTTP/1.1" 500 Internal Server Error
INFO:     127.0.0.1:51761 - "GET /litellm/api/model/info HTTP/1.1" 200 OK

**Comment by start-life** [comment]
It would be great if this could work completely offline.

**Comment by john911f** [comment]
I am using Windows 10. Running ollama from Command Prompt works completely offline, but not working on Open-webui.

Does anyone have any clue to this? I suspect it's some SSL issue.

**Comment by natlamir** [comment]
Try adding this environment variable to the user variables:

Variable name: OLLAMA_BASE_URLS
Variable value: http://127.0.0.1:11434

Then relaunch the conda prompt and try again.

**Reply by Solenhal** [reply]
THANK YOU. This worked for me!

**Reply by anhbsn** [reply]
Please let me know what is "user variables" here? What its mean?
If I want to do it, I have to vi ~/.bashrc >> export OLLAMA_BASE_URLS="http://127.0.0.1:11434" >> source ~/.bashrc? Right? Thankyou!

**Reply by Solenhal** [reply]
_System > Advanced System Settings > Environment Variables > User Variables_

Follow above instructions Natlamir provided, and if they don't exist already, also add:

```
Variable name: OLLAMA_HOST
Variable value: 0.0.0.0:8080
```

And

```
Variable name: OLLAMA_MODELS
Variable value: (Directory where you're storing your models and stuff)
```

Not having these made Ollama unusable. They weren't created automatically so I had to add them manually.

Hope this helped! :D

**Reply by anhbsn** [reply]
Thankyou for your reply, but Im running it on Linux Server, so I just can use command line to communicate with system. Do you have any idea, thanks!

**Reply by KNIGHTABDO** [reply]
so i installed openwebui using pip and i am using this bat script to lunch it offline but its not working for me can u help me figure out what i am missing @echo off
start "Ollama Server" cmd /k "ollama serve"
timeout /t 5 /nobreak >nul
set OFFLINE_MODE=true
set ENABLE_OPENAI_API=false
set ENABLE_GEMINI_API=false
set ENABLE_CLAUDE_API=false
set ENABLE_GOOGLE_GENAI_API=false
set ENABLE_MISTRAL_API=false
set ENABLE_REPLICATE_API=false
set ENABLE_STABILITY_API=false
open-webui serve
pause

**Reply by rgaricano** [reply]
and logs?
Also you can set ollama connection as env var OLLAMA_BASE_URL="http://127.0.0.1:11434"

**Reply by networkMaxi** [reply]
i put all 3 of these user variables in and i got nothing from it. I'm new here just trying to figure out how to use my ollama models in web ui offline. Not working yet, any ideas. I installed using pip its on my local laptop and i have lots of ollama models.
